constraints.pro

```diff
@@ -0,0 +1,105 @@
+%===============================================================================
+% Utility predicates
+%===============================================================================
+
+% True if and only if WorkspaceBasename can unify with the part of the given
+% workspace directory name that results from removing all leading directories.
+workspace_basename(WorkspaceCwd, WorkspaceBasename) :-
+  atomic_list_concat(Parts, '/', WorkspaceCwd),
+  last(Parts, WorkspaceBasename).
+
+% True if and only if WorkspacePackageName can unify with the name of the
+% package which the workspace represents (which comes from the directory where
+% the package is located). Assumes that the package is not in a sub-workspace
+% and is not private.
+workspace_package_name(WorkspaceCwd, WorkspacePackageName) :-
+  workspace_basename(WorkspaceCwd, WorkspaceBasename),
+  atom_concat('@metamask/', WorkspaceBasename, WorkspacePackageName).
+
```

Comment:
It seems that these utility predicates were only used for monorepo-related things (e.g. ensuring the package directory matches the package name). They don't seem to be used here.

```suggestion
```

Reply:
You're right. I tried removing them initially, but something was depending on them, but I think I removed that later. Fixed it now!